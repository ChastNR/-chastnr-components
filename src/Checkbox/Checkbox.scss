.cb {
  border-radius: 8px;
  cursor: pointer;
  display: inline-block;
  height: 24px;
  line-height: 24px;
  position: relative;
  user-select: none;
  width: 24px;

  .cb__input {
    -moz-appearance: none;
    -webkit-appearance: none;
    appearance: none;
    background-color: #ecf1f7;
    border-radius: inherit;
    border: 1px solid transparent;
    cursor: inherit;
    height: 100%;
    left: 0;
    margin: 0;
    outline: none;
    padding: 0;
    position: absolute;
    right: 0;
    top: 0;
    transition-duration: 0.2s;
    transition-property: background-color;
    width: 100%;

    &:checked {
      background-color: #ffdd2d;

      & + span {
        opacity: 1;
        transform: translateY(0);
      }
    }
  }

  & > span {
    display: block;
    opacity: 0;
    pointer-events: none;
    transform: translateY(-4px);
    transition-duration: 0.2s;
    transition-property: transform, opacity;

    & > span {
      align-items: center;
      display: inline-flex;
      justify-content: center;
      position: relative;

      & > span {
        color: inherit;
        display: inline-block;
        height: 24px;
        position: relative;
        transform: none;
        transition-duration: 0.2s;
        transition-property: background-color, color;
        transition-timing-function: ease;
        width: 24px;
      }
    }
  }
}
