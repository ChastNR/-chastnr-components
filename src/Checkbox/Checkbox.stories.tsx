import { useState } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta } from '@storybook/react';
import { Checkbox } from './Checkbox';

const componentMeta: ComponentMeta<typeof Checkbox> = {
  title: 'Checkbox',
  component: Checkbox,
  parameters: {
    layout: 'fullscreen',
  },
};

export const DefaultCheckbox = () => {
  const [isChecked, setChecked] = useState(false);

  const handleChange = () => setChecked((s) => !s);

  return <Checkbox checked={isChecked} onChange={handleChange} />;
};

export default componentMeta;
