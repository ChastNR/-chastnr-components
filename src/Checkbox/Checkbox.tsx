import classNames from 'clsx';
import { ReactComponent as CheckboxSvg } from './checkbox.svg';

import './Checkbox.scss';

interface ICheckboxProps {
  className?: string;
  checked?: boolean;
  disabled?: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Checkbox: React.FC<ICheckboxProps> = ({
  className,
  checked = false,
  disabled,
  onChange = () => {},
}) => {
  const checkBoxClassNames = classNames('cb', className);

  return (
    <div className={checkBoxClassNames}>
      <input
        checked={checked}
        className="cb__input"
        disabled={disabled}
        onChange={onChange}
        type="checkbox"
      />
      <span>
        <span>
          <span>
            <CheckboxSvg />
          </span>
        </span>
      </span>
    </div>
  );
};
