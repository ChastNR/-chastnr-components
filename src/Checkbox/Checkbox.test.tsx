import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Checkbox from "./Checkbox";

describe("<Checkbox/>", () => {
  it("should match snapshot", () => {
    const checkbox = shallow(<Checkbox />);
    expect(toJson(checkbox)).toMatchSnapshot();
  });

  it("props should be equal", () => {
    const checked = true;
    const disabled = true;
    const onChange = jest.fn();

    const checkbox = mount(<Checkbox checked={checked} disabled={disabled} onChange={onChange} />);
    const input = checkbox.find("input");
    input.simulate("change");

    const props = input.props();

    expect(onChange).toBeCalledTimes(1);
    expect(props.checked).toEqual(checked);
    expect(props.disabled).toEqual(disabled);
  });
});
