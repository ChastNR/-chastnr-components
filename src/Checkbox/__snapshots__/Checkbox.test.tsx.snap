// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> should match snapshot 1`] = `
<div
  className="cb"
>
  <input
    className="cb__input"
    type="checkbox"
  />
  <span>
    <span>
      <span>
        <ReactComponent />
      </span>
    </span>
  </span>
</div>
`;
